packages/twenty-server/src/engine/api/graphql/workspace-query-runner/workspace-query-runner.service.ts

```diff
@@ -363,10 +371,12 @@ export class WorkspaceQueryRunnerService {
     args: UpdateManyResolverArgs<Record>,
     options: WorkspaceQueryRunnerOptions,
   ): Promise<Record[] | undefined> {
-    const { workspaceId, objectMetadataItem } = options;
+    const { userId, workspaceId, objectMetadataItem } = options;
 
     assertMutationNotOnRemoteObject(objectMetadataItem);
-    assertIsValidUuid(args.data.id);
+    for (const record of args.data) {
+      assertIsValidUuid(record.id);
```

Comment:
This was already wrong before but we don't want to check the id from data here but from the filter object. In fact, we should reject / not return id in the data type because we don't want to allow update of the ids in an updateMany (nor in an updateOne but for the updateMany won't even work due to unicity constraint)
FYI @charlesBochet @magrinj